workflow/controller/controller.go

```diff
@@ -575,39 +563,18 @@ func (wfc *WorkflowController) processNextPodItem() bool {
 		log.Warnf("watch returned pod unrelated to any workflow: %s", pod.ObjectMeta.Name)
 		return true
 	}
-	// TODO: currently we reawaken the workflow on *any* pod updates.
-	// But this could be be much improved to become smarter by only
-	// requeue the workflow when there are changes that we care about.
-	wfc.wfQueue.Add(pod.ObjectMeta.Namespace + "/" + workflowName)
+	wfc.metrics.PodProcessed()
+	wfc.wfQueue.AddAfter(pod.ObjectMeta.Namespace+"/"+workflowName, 1*time.Second)
```

Comment:
here we throttle workflow changes by delaying them by the same 1s we remove below - we could change this to other values, e.g. 3s

Reply:
Elegant solution!

Reply:
On further thought, perhaps a better solution to this problem would be to only throttle Pods that we can determine were recently created. Perhaps with a tolerance window of a couple of seconds.

If I understand correctly, the issue is that we want to avoid immediately re-queuing a Workflow after creating new pods for it, if so we should only be throttling pods that were recently created, as opposed to all Pod updates. We wouldn't want to delay pod updates that are actually signify something has changed to us. Please correct me if I'm wrong.